### PR TITLE
PP-539 Add all event to the queue

### PR DESCRIPTION
**Pull request by mrkulik**
PP-539

### CODE REVIEW

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift [hunk 1]
```diff
@@ -95,30 +95,15 @@ public class AnalyticsManager {
...
-            eventsQueue.append(queuedEvent)
-            return
+        let queuedEvent = QueuedAnalyticsEvent(event: event,
+                                               screenNameString: screenNameString,
+                                               properties: properties)
+        eventsQueue.append(queuedEvent)
+
+        // Process the event queue if SDKs are initialized
+        if mixpanelInstance != nil && amplitudeInitialised {
+            processEventsQueue()
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:105]: [clarification] Do we still to call this method in `    private static func handleAnalyticsSDKsInit() {` ? I think is not neccessary.

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift [hunk 2]
```diff
@@ -136,9 +121,22 @@ public class AnalyticsManager {
     private static func processEventsQueue() {
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:121]: [suggestion] I think we should rename this method. What do you think about `trackEventsFromQueue` or `trackQueueEvents`?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift [hunk 3]
```diff
@@ -136,9 +121,22 @@ public class AnalyticsManager {
...
+            for property in queuedEvent.properties {
+                let propertyValue = property.value.analyticsPropertyValue()
+                eventProperties[property.key.rawValue] = convertPropertyValueToString(propertyValue)
+            }
+
+            // Track event in Mixpanel
+            mixpanelInstance?.track(event: queuedEvent.event.rawValue, properties: eventProperties)
+
+            // Track event in Amplitude
+            ampltitudeTrackEvent(event: queuedEvent.event, eventProperties: eventProperties)
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:139]: [suggestion] Maybe we can put this portion of code into a separate method. What do you think?

Comment (mrkulik) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:139]: @ValentinaIancu-Gini 
I think we can put this code in method track(event: QueuedAnalyticsEvent) and call it from `processEventsQueue`. In that way we will keep consistent structure of our methods
Works good?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift (hunk 2 above)
Comment (mrkulik) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:121]: @ValentinaIancu-Gini if we will move out tracking events code from this method and I suggest to choose something from this:
`releaseEventsQueue`
`processEventsQueue`
`flushEventsQueue`

Comment (ValentinaIancu-Gini) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Tracking/AnalyticsManager.swift:121]: mhh let's keep it as it is`processEventsQueue` but maybe we can add a comment to this method and describe what is actually doing.

**Review by ValentinaIancu-Gini — APPROVED** [review]
@mrkulik good job.
Let's wait for the checks to pass and after that, we can merge this.
Thank you!

**Pull request merged**